pages/dev/solidity-utilities.md

```diff
@@ -0,0 +1,39 @@
+# Solidity Utilities
+
+To ease cross chain developement we have provided some solidity utilites. These can be found [here](https://github.com/axelarnetwork/axelar-utils-solidity). Each of these as well as their proposed use is detialed below.
+
+## Constant Address Deployer
+
+Creating a cross chain dApp will often require the same contract to be deployed on multiple chains. Furthermore it is usefull to know each address of this contract on each chain, either to know where to send remote contract calls, or where to trust remote contract calls from, often both. If we can guarantee that the contracts in question will be deployed at the same address on each network then the above is trivial. This can be achieved by deploying each contract from the same address with the same nonce at each network, or by using [`create2`](https://eips.ethereum.org/EIPS/eip-1014). For this purpose we deployed [`ConstAddressDeployer`](https://github.com/axelarnetwork/axelar-utils-solidity/blob/main/src/ConstAddressDeployer.sol) at `0x617179a15fEAa53Fa82ae80b0fc3E85b7359a748` on every EVM testnet and mainnet that is supported by Axelar, and we plan on deploying it on future supported testnets and mainnets too. `ConstAddressDeployer` exposes the following functions
```

Comment:
```suggestion
Creating a cross chain dApp will often require the same contract to be deployed on multiple chains. Furthermore it is useful to know each address of this contract on each chain, either to know where to send remote contract calls, or where to trust remote contract calls from, often both. If we can guarantee that the contracts in question will be deployed at the same address on each network then the above is trivial. This can be achieved by deploying each contract from the same address with the same nonce at each network, or by using [`create2`](https://eips.ethereum.org/EIPS/eip-1014). For this purpose we deployed [`ConstAddressDeployer`](https://github.com/axelarnetwork/axelar-utils-solidity/blob/main/src/ConstAddressDeployer.sol) at `0x617179a15fEAa53Fa82ae80b0fc3E85b7359a748` on every EVM testnet and mainnet that is supported by Axelar, and we plan on deploying it on future supported testnets and mainnets too. `ConstAddressDeployer` exposes the following functions
```